src/Admin/Exports/Services/YouTubeOrders.php

```diff
@@ -0,0 +1,121 @@
+<?php
+declare( strict_types=1 );
+
+namespace Automattic\WooCommerce\GoogleListingsAndAds\Admin\Exports\Services;
+
+use Automattic\WooCommerce\GoogleListingsAndAds\Infrastructure\Service;
+
+defined( 'ABSPATH' ) || exit;
+
+/**
+ * Class YouTubeOrders
+ *
+ * @package Automattic\WooCommerce\GoogleListingsAndAds\Admin\Exports\Contracts
+ */
+class YouTubeOrders implements Service {
+	/**
+	 * Return an array of WooCommerce order IDs for the Merchant Conversions Report CSV.
+	 *
+	 * @param string  $date
+	 * @param integer $limit
+	 * @param integer $offset
+	 * @return array
+	 */
+	public function find_orders( string $date = '', int $limit = -1, int $offset = 0 ): array {
+		// Use yesterdays date if no date passed.
+		$date = empty( $date ) ? gmdate( 'Y-m-d', strtotime( '-1 day' ) ) : $date;
+
+		// Fetch all orders and refunds for the date without meta filtering.
+		// We'll filter by meta during the loop to leverage meta caching.
+		$purchase_ids = $this->find_purchases( $date, $limit, $offset );
+		$refund_ids   = $this->find_refunds( $date, $limit, $offset );
+
+		// Filter orders and refunds by YouTube attribution meta.
+		$filtered_order_ids = [];
+		$refund_parent_ids  = [];
+
+		// Filter refunds first to collect parent order IDs
+		foreach ( $refund_ids as $refund_id ) {
+			$refund = wc_get_order( $refund_id );
+			if ( $refund && 'shop_order_refund' === $refund->get_type() ) {
```

Comment:
@jamesmorrison when checking if an order is a refund we may want to check the `$order` instance instead of `get_type()` like we do in the `OrderItemRowBuilder`

`if ( $order instanceof WC_Order_Refund )`